### DISCUSSION TITLE
Path selection for storage adapters ie AWS

### DISCUSSION THREAD

**Discussion by pandora-otto** [directus/directus#7092] [Archive]
Hello guys

I'm trying to make the file uploading more flexible; to define somehow the path/folder of the image to be saved to.

I have setup .env to use the AWS storage adapter. I know that in the bucket parameter 'STORAGE_AWS_BUCKET' I can include the path too, in order to save files in a specific folder. But how do I set it up so I can use multiple folders?

Example.
Say I have a collection 'slideshows' and I want to upload images in a folder 'slideshows' in my S3 bucket, ie /mybucket/images/slideshows.

I also have a collection 'articles' and I'd like to upload an image in the following path /mybucket/images/articles.

Would it be possible to add a field in these collections something like 'remote_path' and use this field during upload process?

So in my 'slideshows' collection I create a 'remote_path' field and equal to '/images/slideshows' and in my 'images/articles' collection I create a 'remote_path' field and equal to '/images/articles' etc.

If that's doable, we could take it a step further and create a collection of 'remote_paths' with all of our folders that we want to use. Then each of our collections 'slideshows', 'articles' etc can have a Many-2-One field with 'remote_paths' and select a path each time we want to upload an image.

Lastly, I read about the Multiple Storage Adapters https://docs.directus.io/guides/api-config/#multiple-storage-adapters .
I don't entirely comprehend the way to use CSV of storage location names. Is there an example somewhere to see, I couldn't fine one. Could this way be the solution I'm looking for?

As always, many thanks in advance and keep up the good work you guys do!

**Comment by rijkvanzanten** [comment]
@pandora-otto Are you looking to have the folders matched to the storage adapter, or looking to setup multiple root locations outside of Directus' (virtual) folders?

**Reply by pandora-otto** [reply]
I think a bit of both!?

I'd like to deal with aws s3 when it comes for images and files. I have already images in s3 in various folders. I would like to be able to upload and delete the images there using Directus. Now depending on what 'collection' I use, when I upload an image, to have a predefined folder set so that it is placed there.

As in my first post, if I upload a file for my collection 'articles' the image should go to /bucket/images/articles etc.
Right now all images go to whatever I put in 'STORAGE_AWS_BUCKET' parameter. Not ideal.

I hope my description of what I want to achieve is good, if not I can elaborate more, perhaps with screenshots too.

Thank you @rijkvanzanten

**Reply by rijkvanzanten** [reply]
Yeah! You're use case makes total sense, I'm just trying to think through the backend to see what approach would make most sense from an implementation detail 🙂 

It sounds like it would be ideal if there was a way to enable "folder syncing" to the storage adapter, so every time you make a virtual folder in directus, an actual folder path is used in s3 as well. This in turn would allow you to use the option on the image interface to upload to a specific folder, which would allow you to configure your articles image field to upload to the `/path/to/your/articles` folder on s3 🤔 (cc @benhaynes)

**Reply by pandora-otto** [reply]
This sounds so ideal and proper.
Surely this would make other Directus users happy! :100: :dancers: 

Thank you @rijkvanzanten
